### PR TITLE
Ensure fetched langpacks have a summary, categories and license.

### PR DESCRIPTION
**Pull request by eviljeff**

### CODE REVIEW

File: src/olympia/zadmin/tasks.py [hunk 1]
```diff
@@ -587,14 +587,30 @@ def fetch_langpack(url, xpi, **kw):
             log.info('[@None] Created new "{0}" language pack, version {1}'
                      .format(xpi, data['version']))
 
+        # Set the category
+        for app in version.compatible_apps:
+            static_category = (
+                CATEGORIES[app.id][amo.ADDON_LPAPP]['general'])
```
Comment (EnTeQuAk) [inline src/olympia/zadmin/tasks.py]: Just making sure… Android doesn't have any `ADDON_LPAPP` key defined, so if this is a real scenario (which I don't know) we might want to add a check here.

File: src/olympia/zadmin/tasks.py [hunk 2]
```diff
@@ -587,14 +587,30 @@ def fetch_langpack(url, xpi, **kw):
...
+            category, _ = Category.objects.get_or_create(
+                id=static_category.id, defaults=static_category.__dict__)
+            AddonCategory.objects.get_or_create(addon=addon, category=category)
+        del addon.all_categories
```
Comment (EnTeQuAk) [inline src/olympia/zadmin/tasks.py]: Let's add a comment like `# Remove old, outdated categories cache on the model.` or so. It's not immediately clear that `all_categories` is just simply a cache.

File: src/olympia/zadmin/tasks.py [hunk 3]
```diff
@@ -587,14 +587,30 @@ def fetch_langpack(url, xpi, **kw):
...
         file_ = version.files.get()
         if not is_beta:
             # Not `version.files.update`, because we need to trigger save
             # hooks.
             file_.update(status=amo.STATUS_PUBLIC)
         sign_file(file_, settings.SIGNING_SERVER)
 
-        addon.update_version()
+        addon.update(status=amo.STATUS_PUBLIC,
+                     summary=(addon.summary if addon.summary else addon.name))
```
Comment (EnTeQuAk) [inline src/olympia/zadmin/tasks.py]: why do we set the summary here? Is this a fall-back for a potentially missing summary and the field being not nullable? I don't see a test for this.

File: src/olympia/zadmin/tasks.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/zadmin/tasks.py]: Android doesn't have any [langpacks](https://addons.mozilla.org/en-US/android/language-tools/).  For now we could just ignore the categories in the unlikely scenario one appears, but the point of the patch is to create complete add-ons that can be made public - when that work is finished the add-on can't be public so things are going to stop working, to some extent at least.

File: src/olympia/zadmin/tasks.py (hunk 3 above)
Comment (eviljeff) [inline src/olympia/zadmin/tasks.py]: It's a fallback for potentially missing summary, yes.  `Addon.summary` is nullable, but a summary is one of the pieces of meta-data required to be 'complete' and eligible to be publicly available.
I'll add a test.

Comment (EnTeQuAk) [inline src/olympia/zadmin/tasks.py]: Perfect, thanks!

File: src/olympia/zadmin/tasks.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/zadmin/tasks.py]: https://github.com/mozilla/addons-server/pull/4761/commits/e313829f803138cf2d6d6e6015107aea15e73895

**Pull request merged**